hibernate-core/src/main/java/org/hibernate/cfg/schema/SchemaNamingProviderLocator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.cfg.schema;
+
+import org.hibernate.annotations.common.reflection.XClass;
+
+/**
+ * SchemaNamingProvider singleton locator.<br>
+ *
+ * TODO : integrate into class AnnotationSettings as "official" configuration. <br>
+ * TODO (later) : migrate into a non static approach... (complex)
+ *
+ * @author Benoit Besson
+ */
+public class SchemaNamingProviderLocator {
+
+	/** static instance to delegate on. */
+	private static SchemaNamingProvider instance;
```

Comment:
A static variable without any thread safety blocks is risky. Why not use a local variable and wrap it inside an object that confined to the caller.